### PR TITLE
feat: add --inclusive-tpot output flag

### PR DESCRIPTION
**Pull request by natoscott**
## Summary

AIC's TPOT metric is inter-token latency (decode phase only). This PR adds `--inclusive-tpot`, a flag that reports TPOT as `(ttft + tpot * (osl - 1)) / osl` at output time, spreading TTFT cost across all output tokens. This matches the end-to-end per-token latency definition used by GuideLLM and other benchmarking tools, making predicted values directly comparable to benchmark measurements.

- Transformation applied to terminal output and saved CSV only — internal DataFrames, SLA filtering, and pareto computation are unaffected
- Available in `default` and `exp` modes
- The `estimate` subcommand has a separate single-point output path reading scalar attributes directly and is not affected by this flag
- The Python SDK (`cli_default`/`cli_exp`) does not expose the flag; SDK callers receive raw DataFrames and can apply the formula themselves if needed

## Changes

- `src/aiconfigurator/cli/main.py` — add `--inclusive-tpot` to `default` and `exp` subparsers; thread `inclusive_tpot` through `_execute_task_configs`
- `src/aiconfigurator/cli/report_and_save.py` — add `_apply_inclusive_tpot` helper; apply at entry of `log_final_summary` and `save_results`
- `docs/cli_user_guide.md` — document flag in defaults table and add section near `--strict-sla`
- `tests/unit/cli/test_cli_workflow.py` — unit tests for formula correctness, no-mutation, TTFT unchanged, missing columns, and osl=1 boundary case
- `tests/unit/cli/test_argument_parsing.py` — assert default is False

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added a CLI flag to optionally report per-output-token latency by distributing startup time across tokens.

* **Documentation**
  * User guide updated with flag details and examples of reporting behavior.

* **Behavioral Notes**
  * Terminal output, CSV exports, and Pareto plots show inclusive TPOT when enabled; SLA filtering, worker setup, and deployment artifacts continue to use the original inter-token metric.

* **Tests**
  * Unit tests added/updated to cover flag parsing and inclusive TPOT computation.

[Review Change Stack](https://app.coderabbit.ai/change-stack/ai-dynamo/aiconfigurator/pull/1141?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/aiconfigurator/cli/report_and_save.py [hunk 1]
```diff
@@ -284,8 +296,20 @@ def log_final_summary(
...
+    # display_* copies carry inclusive TPOT for printed values only.
+    # The originals are kept for _plot_worker_setup_table which does TPOT filtering.
+    if inclusive_tpot:
+        display_best_configs = {k: _apply_inclusive_tpot(v) for k, v in best_configs.items()}
+        display_pareto_fronts = {
+            k: _apply_inclusive_tpot(v) if v is not None else None for k, v in pareto_fronts.items()
+        }
+    else:
+        display_best_configs = best_configs
+        display_pareto_fronts = pareto_fronts
```
[the remark this answers is not in the document]
Comment (natoscott) [reply src/aiconfigurator/cli/report_and_save.py:311]: @coderabbitai should be resolved now, please review again

File: src/aiconfigurator/cli/report_and_save.py [hunk 2]
```diff
@@ -28,6 +28,18 @@
...
+def _apply_inclusive_tpot(df: pd.DataFrame) -> pd.DataFrame:
+    """Return a copy of df with tpot replaced by inclusive semantics.
+
+    inclusive tpot = (ttft + tpot * (osl - 1)) / osl
+    Applied to output copies only — never to internal computation DataFrames.
+    """
+    df = df.copy()
+    if "ttft" in df.columns and "tpot" in df.columns and "osl" in df.columns:
+        df["tpot"] = (df["ttft"] + df["tpot"] * (df["osl"] - 1)) / df["osl"]
+    return df
```
[the remark this answers is not in the document]
Comment (Arsene12358) [reply src/aiconfigurator/cli/report_and_save.py]: osl will never be zero in the context of LLM inference

[the remark this answers is not in the document]
Comment (natoscott) [reply src/aiconfigurator/cli/report_and_save.py]: @Arsene12358  I see, yes - osl is always ≥ 1 (minimum one output token), so division by zero can't occur. The column existence guard I added also didn't address coderabbit's actual concern; removing it entirely is cleaner - I'll do that.

**Review by jasonqinzhou — APPROVED** [review]
Please fix the test failures and resolve comments.

**Review by jasonqinzhou — APPROVED** [review]

**Comment by natoscott** [comment]
> Please fix the test failures and resolve comments.

Ah, thanks - the test failures were introduced by overly-optimistic changes I made based on the initial review - we do need a missing-column guard for ttft with the way the tests invoke this code.  Fixed.

I think all other comments are resolved - please let me know if I missed anything in particular though.  The failing pre-check on docstrings coverage is pre-existing and project-wide (this PR adds one function — _apply_inclusive_tpot — and it has a docstring describing it).
